### PR TITLE
Document JSONBench column-store ClickHouse experiment

### PR DESCRIPTION
**Pull request by snissn**
## Objective

Adds a technical experiment writeup for the preferred/server-shaped 10M TreeDB column-store JSONBench comparison against ClickHouse.

## Context

Provides a reproducible benchmark record comparing TreeDB's production column-store path against ClickHouse JSON on a local 10M-row Bluesky fixture, establishing a baseline for future performance tracking.

## Non-goals

- Not a product claim or official performance benchmark.
- Not adding new code, APIs, or runtime behavior — documentation only.

## Design

New doc:

- `TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md`

The document covers:

- scope and caveats for the experiment;
- the single reproduction entry point in JSONBench (merge commit `9a60981712a5c685ebc0c8e059f65eb022b55251`);
- exact prerequisites and output artifacts (referenced as `$OUT_DIR`-relative paths): `preferred_summary.md`, `treedb/report.md`, `treedb/report.json`, `clickhouse/result.json`;
- what TreeDB rows are timed as prepared physical runners vs aggregate metadata, with `*` markers on q1/q2/q3 relabeled rows and a footnote for the layout-key rename;
- current 10M TreeDB vs ClickHouse results (i5-11400F, 6C/12T, 31GiB RAM) with per-attempt timing tables for both systems;
- throughput column labeled `effective query rows/s`; q5 normalized to `1,017.9M logical`;
- ClickHouse scanned rows using the actual loaded count (`9,999,994`) throughout the results table;
- storage/load caveats and follow-ups.

Companion JSONBench PR with the single-entry runner: https://github.com/snissn/JSONBench/pull/6

## Scope

- Includes:
  - `TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md` (new)
  - `TreeDB/docs/guides/README.md` (adds link to new guide)
- Excludes: all runtime code, tests, benchmarks, and other docs.

## Correctness

- Tests run (exact commands):
  ```sh
  GOWORK=off go test ./TreeDB/docs -count=1
  go test ./TreeDB/collections -run TestTypedStorageLegacyNameAllowlistIsComplete
  ```
- Corruption/edge cases covered: N/A (documentation only).
- Concurrency/race coverage: N/A.

## Performance

- Benchmarks run: N/A (this PR documents an existing run, not adds a new benchmark).
- Run host: i5-11400F, 6 cores / 12 threads, 31 GiB RAM, Ubuntu 22.04, kernel `6.8.0-110-generic`.
- JSONBench benchmark commit for captured run: `5fca4d68f3f06909de0cc87b2f012bc3583a3170`; single-entry reproduction wrapper added in merge commit `9a60981712a5c685ebc0c8e059f65eb022b55251`.
- gomap: `29dd3cb3857baff8b7889c21b8be30957f9e9bc2`.
- ClickHouse: `26.4.2.10`.
- Rows: TreeDB 10,000,000; ClickHouse 9,999,994 loaded from 10,000,000 requested.
- Summary artifact: `preferred_summary.md` (under `$OUT_DIR`), matching the merged JSONBench runner output.

Primary results are included in the markdown doc.

## Rollout / Toggle Plan

- Default setting: N/A (documentation only).
- How to disable quickly: N/A.

## Follow-ups

- Track benchmark-shape cleanup in gomap issue #1889: introduce an explicit `column-store-prepared` JSONBench layout instead of relabeling q1/q2/q3 in summaries.
- Keep direct one-shot rows separate from server-shaped prepared rows in future JSONBench results.

## Column Graph Native Workstream (#1646, if applicable)

N/A — documentation only.

### AI Review Loop

- Copilot latest-head review: all review threads resolved; `156bbc9d0` confirmed clean.
- Review comments/threads resolved or explicitly dismissed: all threads resolved across multiple review passes (artifact naming, identifying info, ClickHouse scanned-row denominator, per-attempt tables, throughput header).
- Re-requested after final meaningful push: yes.

---

### Scope
- [x] Single, clear objective for this PR (not a bundle)
- [x] Explicit include list (files/symbols touched): `TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md`, `TreeDB/docs/guides/README.md`
- [x] Explicit exclude list (files/symbols NOT touched): all runtime code
- [x] No unwanted feature reintroduction

### Safety / Correctness
- [x] No code changes; documentation only

### Tests
- [x] Ran locally:
  - [x] `GOWORK=off go test ./TreeDB/docs -count=1`
  - [x] `go test ./TreeDB/collections -run TestTypedStorageLegacyNameAllowlistIsComplete`

### Benchmarks / Measurements
- [x] Reported results in PR description (run host hardware, versions, row counts)

### Docs
- [x] New guide added with reproduction entry point, prerequisites, results table, per-attempt timing tables, and caveats
- [x] Artifact names consistent throughout document (`preferred_summary.md`, matching merged JSONBench runner)
- [x] ClickHouse scanned rows use actual loaded count (`9,999,994`) throughout results table
- [x] No identifying username/IP/hostname in public doc; CPU/RAM/OS bullets capture reproducibility context

### CODE REVIEW

**Comment by snissn** [comment]
@codex review
@copilot review
@coderabbitai review

**Comment by snissn** [comment]
Updated the doc in latest head `da7984b67` after review:\n\n- notes the JSONBench reproduction runner merge commit;\n- restores the actual new script output name `preferred_summary.md`;\n- renames `loaded rows/s` to `query rows/s`;\n- normalizes q5 throughput to `1,017.9M logical`;\n- adds TreeDB per-attempt timings to match the ClickHouse attempts table;\n- adds a footnote clarifying that q1/q2/q3 are relabeled from the current JSONBench layout key.\n\nRe-ran:\n\n```sh\ngo test ./TreeDB/docs\n```\n\n@codex review\n@copilot review\n@coderabbitai review

**Comment by snissn** [comment]
Pushed follow-up `c58b3ddfb` for the remaining doc review nits:\n\n- uses visible `*` markers on q1/q2/q3 relabeled rows;\n- changes the throughput header to `effective query rows/s`;\n- keeps q5 in M-units as `1,017.9M logical`;\n- preserves the TreeDB attempts table added in the prior update;\n- removes duplicate relabeling prose.\n\nRe-ran:\n\n```sh\ngo test ./TreeDB/docs\n```\n\n@codex review\n@copilot review\n@coderabbitai review

**Comment by snissn** [comment]
Pushed `e63ad658d` to clarify artifact names consistently with the merged JSONBench runner:\n\n- reproduction artifacts now list `preferred_summary.md`, `treedb/report.md`, `treedb/report.json`, and `clickhouse/result.json`;\n- the historical capture is described as equivalent prototype commands before the wrapper merge.\n\nRe-ran:\n\n```sh\ngo test ./TreeDB/docs\n```

File: TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# Re-render/reuse an existing half of a run.
+RUN_TREEDB=0 ./run_preferred_columnstore_clickhouse_compare.sh
+RUN_CLICKHOUSE=0 ./run_preferred_columnstore_clickhouse_compare.sh
+```
+
+Primary outputs:
+
+- `preferred_summary.md` — compact TreeDB-vs-ClickHouse table.
+- `treedb/report.md`, `treedb/report.json` — TreeDB JSONBench report.
+- `clickhouse/result.json` — ClickHouse JSONBench-compatible result.
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md:58]: Kept `preferred_summary.md` intentionally: that is the artifact emitted by the merged JSONBench reproduction wrapper (`SUMMARY_OUT="$OUT_DIR/preferred_summary.md"`). The doc is now aligned with the actual script output.

File: TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+- TreeDB rows loaded: 10,000,000.
+- ClickHouse rows loaded: 9,999,994 of 10,000,000 requested.
+- Tries: 3 query attempts.
+
+Reproduction artifacts are written under `$OUT_DIR`:
+
+- `preferred_summary.md`
+- `treedb/report.md`
+- `treedb/report.json`
+- `clickhouse/result.json`
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md:110]: Kept `preferred_summary.md` intentionally: it matches the merged JSONBench runner output and the Primary outputs section. The older `preferred_10m_treedb_clickhouse_summary.md` name was from the prototype capture, not the reproduction entry point.

File: TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+| TreeDB `column-store-prepared` | q1 | 0.0122s | 822.9M | 10.0M | 1.11 GiB | 54.754s | 2.1x faster |
+| ClickHouse JSON | q1 | 0.0260s | 384.6M | 10.0M | 986.82 MiB | 52.632s | baseline |
+| TreeDB `column-store-prepared` | q2 | 0.0518s | 192.9M | 10.0M | 3.21 GiB | 105.283s | 3.8x faster |
+| ClickHouse JSON | q2 | 0.1970s | 50.8M | 10.0M | 986.82 MiB | 52.632s | baseline |
+| TreeDB `column-store-prepared` | q3 | 0.1201s | 83.3M | 10.0M | 2.63 GiB | 103.099s | 1.1x slower |
+| ClickHouse JSON | q3 | 0.1060s | 94.3M | 10.0M | 986.82 MiB | 52.632s | baseline |
+| TreeDB `column-store-prepared-metadata` | q4 | 0.0100s | 996.4M logical | 0 | 3.10 GiB | 124.950s | 8.9x faster |
+| ClickHouse JSON | q4 | 0.0890s | 112.4M | 10.0M | 986.82 MiB | 52.632s | baseline |
+| TreeDB `column-store-prepared-metadata` | q5 | 0.0098s | 1,017.9M logical | 0 | 3.10 GiB | 125.455s | 6.8x faster |
+| ClickHouse JSON | q5 | 0.0670s | 149.3M | 10.0M | 986.82 MiB | 52.632s | baseline |
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/docs/guides/jsonbench-columnstore-clickhouse-experiment.md]: Fixed in `156bbc9d0`: ClickHouse scanned rows now use the actual loaded row count (`9,999,994`) in every results-table row, matching the caveat and denominator.

**Comment by snissn** [comment]
Pushed `156bbc9d0` for the latest review feedback:

- ClickHouse rows in the results table now show the actual loaded/scanned row count (`9,999,994`) instead of rounded `10.0M`, matching the caveat and denominator.
- Kept `preferred_summary.md` intentionally because that is the artifact emitted by the merged JSONBench reproduction wrapper.
- Updated/resolved the Copilot threads.

Re-ran:

```sh
go test ./TreeDB/docs
go test ./TreeDB/collections -run TestTypedStorageLegacyNameAllowlistIsComplete
```

@codex review
@copilot review
@coderabbitai review
